sklearn/ensemble/gradient_boosting.py

```diff
@@ -1836,3 +1843,353 @@ def apply(self, X):
+        possbile to make predections later with this class.
+
+    pre_dispatch : int, or string, optional, default='2*n_jobs'
+        Controls the number of jobs that get dispatched during parallel
+        execution. Reducing this number can be useful to avoid an
+        explosion of memory consumption when more jobs get dispatched
+        than CPUs can process. This parameter can be:
+
+            - None, in which case all the jobs are immediately
+              created and spawned. Use this for lightweight and
+              fast-running jobs, to avoid delays due to on-demand
+              spawning of the jobs
+
+            - An int, giving the exact number of total jobs that are
+              spawned
+
+            - A string, giving an expression as a function of n_jobs,
+              as in '2*n_jobs'
+
+    n_jobs : int, default=1
+        Number of jobs to run in parallel.
+
+    pre_dispatch : int, or string, optional
+        Controls the number of jobs that get dispatched during parallel
+        execution. Reducing this number can be useful to avoid an
+        explosion of memory consumption when more jobs get dispatched
+        than CPUs can process. This parameter can be:
+
+            - None, in which case all the jobs are immediately
+              created and spawned. Use this for lightweight and
+              fast-running jobs, to avoid delays due to on-demand
+              spawning of the jobs
+
+            - An int, giving the exact number of total jobs that are
+              spawned
+
+            - A string, giving an expression as a function of n_jobs,
+              as in '2*n_jobs'
+
+    verbose : integer
+        Controls the verbosity: the higher, the more messages.
+
+    Attributes
+    ----------
+    grid_scores_ : list of named tuples
+        Contains scores for all parameter combinations in param_grid.
+        Each entry corresponds to one parameter setting.
+        Each named tuple has the attributes:
+
+            * ``parameters``, a dict of parameter settings
+            * ``mean_validation_score``, the mean score over the
+              cross-validation folds
+            * ``cv_validation_scores``, the list of scores for each fold
+
+    best_score_tuple_ : named tuple
+        The attributes of the model which gave the best score.
+
+        * ``parameters``, a dict of parameter settings, which includes the
+          `n_estimators` parameter
+        * ``mean_validation_score``, the mean score over the
+          cross-validation folds
+        * ``cv_validation_scores``, the list of scores for each fold
+
+    best_params_ : dict
+        The parameters which gave the best score, along with the `n_estimators`
+        parameter.
+
+    best_estimator_ : GradientBoostingClassifier
+        If `refit` was `True`, the model with the best score fit to the entire
+        dataset.
+
+    """
+
+    estimator_class = GradientBoostingClassifier
+
+    def __init__(self, **kwargs):
```

Comment:
Wouldn't that make the docstring needlessly big and complex ? Isn't it better to leave the other parameters untouched and let them be handled by the gradient boosting class ?